src/Setup/BuildPlan.purs

```diff
@@ -78,3 +87,9 @@ resolve versionsContents tool = do
 
         Right v -> do
           pure (pure { tool, version: v })
+
+    Just Unstable -> do
+      liftEffect do
+        Core.info $ fold [ "Fetching most recent version (pre-release or not) for ", Tool.name tool ]
+      version <- liftAff $ fetchFromGitHubReleases AnyRelease (Tool.repository tool)
+      pure (pure { tool, version })
```

Comment:
Usually we read the latest tag from the stored `versions.json` file:
https://github.com/purescript-contrib/setup-purescript/blob/main/dist/versions.json

We do this because it's really easy to burn requests to the GitHub API and these versions don't change that frequently. You can see this in practice in the `Latest` case right above this, where we read the file.

I think we might want to update the format of this file to something like:

```json
{
  "purs": { "stable": ..., "unstable": ... },
  ...
}
```

and then we can read either the latest stable or unstable version depending on what the user wants. Note that if we ever want to update to the latest unstable version faster than the daily check this repo does, we can always manually update the `versions.json` file right away.

Reply:
(At this point it would probably help to have an actual type and JSON encoding / decoding, rather than the inline thing we've got going now)

Reply:
I keep trying to write this but then run into friction over the design of this. Could you clarify why `Tool` and `Key` are separate things?

Reply:
The difference is a little annoying, and we probably should have just used the executable name for everything. But the reason it exists is that there can be a difference between the name of a tool (PureScript) and the name of the executable (purs).

Other than that, the only difference between Tool and Key is that the tool is a data type and the key is a string, the actual thing the user types into their action.yml file.